### PR TITLE
#45 Use Unsafe instantiator on JDK9

### PR DESCRIPTION
**Pull request by ropalka**
Proposal solution for issue https://github.com/easymock/objenesis/issues/45

### CODE REVIEW

**Comment by ropalka** [comment]
When this issue is resolved it would be great to have org.objenesis:objenesis:2.5 released :)

**Comment by henri-tremblay** [comment]
Thanks. I should be able to have a look and release in about a week

File: main/src/main/java/org/objenesis/strategy/StdInstantiatorStrategy.java [hunk 1]
```diff
@@ -62,9 +62,14 @@
             }
             return new AccessibleInstantiator<T>(type);
          }
-         // The UnsafeFactoryInstantiator would also work. But according to benchmarks, it is 2.5
-         // times slower. So I prefer to use this one
-         return new SunReflectionFactoryInstantiator<T>(type);
+         if (PlatformDescription.VM_VERSION.startsWith("9")) {
+            // UnsafeFactoryInstantiator doesn't work on JDK9 anymore
```
Comment (jcarvalho) [inline main/src/main/java/org/objenesis/strategy/StdInstantiatorStrategy.java]: Isn't this comment contradictory? '`UnsafeFactoryInstantiator` doesn't work in JDK9, so let's use it!'

Comment (henri-tremblay) [inline main/src/main/java/org/objenesis/strategy/StdInstantiatorStrategy.java]: Indeed. Is it Unsafe or ReflectionFactory that doesn't work?

Comment (ropalka) [inline main/src/main/java/org/objenesis/strategy/StdInstantiatorStrategy.java]: Ouch, copy-paste error :( Of course comment should be:
// SunReflectionFactoryInstantiator doesn't work on JDK9 anymore

**Comment by ropalka** [comment]
The method will be resuscitated so false alarm, see:
https://bugs.openjdk.java.net/browse/JDK-8168980

**Comment by henri-tremblay** [comment]
Funny. Let's consider this unneeded then. Thanks for the alert, the patch, the investigation and the follow-up

**Pull request closed without merging**